lib/Kaba.js

```diff
@@ -403,20 +403,14 @@ class Kaba
      */
     getBuildConfig (cliConfig)
     {
-        let buildConfig = {
+        return {
             sass: this.buildSassConfig(),
             js: {
                 javaScriptDependenciesFileName: this.javaScriptDependenciesFileName,
                 webpack: this.buildWebpackConfig(cliConfig),
+                customTypeScriptConfig: this.customTypeScriptConfig,
```

Comment:
The parameter definition for this field is missing, so you are currently creating a dynamic class which is generally not a good idea.

Please add it and initialize it to `null`. You need to adjust your type above, then.